pkg/consolegraphql/resolvers/resolver_address.go

```diff
@@ -272,3 +297,47 @@ func tokenizeAddress(name string) ([]string, error) {
 	}
 	return addressToks, nil
 }
+
+func defaultResourceNameFromAddress(input *v1beta1.Address) error {
+	if input.Spec.Address == "" {
+		return fmt.Errorf("address is undefined, cannot default resource name from address '%s'",
+			input.Spec.Address)
+	}
+	if input.Annotations != nil {
+		if as, present := input.Annotations[addressSpaceNameAnnotationKey]; present {
```

Comment:
Is this required to be set by the console and always guaranteed to be correct?

Reply:
As things stand now, the console ui will be required to set the annotation in order to let the backend know which address space the new address belongs.

On reflection, maybe it would be better to avoid exposing the implementational detail (the annotation is really a private concern of the controller) and change the GraphQL API (createAddress, addressCommand) to accept an addressSpaceName parameter.  The parameter would be required if you want the server to default the address resource name.

@lulf @vbusch  WDYT?

Reply:
@k-wall Ok, although putting it in the request is nice, I'm fine with leaving it as annotation too, mainly because we would not have to change the API once we remove address spaces.

Reply:
Having createAddress & addressCommand take the addressSpaceName seems more straight forward.
Soon, with shared infra, the addressSpace will no longer be required, as we will only be using the namespace.